lib/deploy/functions/release.js

```diff
@@ -318,24 +262,44 @@ module.exports = function(context, options, payload) {
       .map(function(functionName) {
         utils.logBullet(chalk.bold.cyan('functions: ') + 'deleting function ' + chalk.bold(functionName) + '...');
         _startTimer(functionName, 'delete');
-        return gcp.cloudfunctions.delete(projectId, GCP_REGION, functionName);
+        deployments.push({
+          functionName: functionName,
+          retryFunction: gcp.cloudfunctions.delete,
+          retryParams: {
+            projectId: projectId,
+            region: GCP_REGION,
+            functionName: functionName
+          }
+        });
       }).value();
-    return RSVP.allSettled([].concat(addOps, updateOps, deleteOps));
+
+    return RSVP.all([].concat(createOps, updateOps, deleteOps)).then(function() {
+      return RSVP.allSettled(_.map(deployments, function(op) {
+        return op.retryFunction(op.retryParams).then(function(res) {
```

Comment:
I think this could be simplified. Instead of passing `retryFunction` and `retryParams`, couldn't you (for each of create/update/delete) do something like:

```js
deployments.push(function() {
  return gcp.cloudfunctions.create({/* ... */});
});
```

Then, here, you could do something like:

```js
return RSVP.allSettled(_.map(deployments, function(op) {
  return op().catch(function(err) {
    return errMatchesRetryableConditions ? op() : RSVP.reject(err);
  });
});
```

With this approach you could probably even avoid `deployments` and continue using a `map` of create, update, and delete ops.